### PR TITLE
13.0 mig website event require login

### PR DESCRIPTION
**Pull request by emagdalenaC2i**

### CODE REVIEW

**Comment by emagdalenaC2i** [comment]
@chienandalu and @pedrobaeza could you review this PR?

**Comment by emagdalenaC2i** [comment]
@chienandalu and @pedrobaeza This is a standard migration, could you review it

**Comment by emagdalenaC2i** [comment]
@Yajo @rafaelbn and @NL66278 Please, could you review this PR?

File: website_event_require_login/controllers/main.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+from odoo.addons.website_event.controllers.main import WebsiteEventController
+
+
+class RequireLoginToRegister(WebsiteEventController):
+    @route()
+    def registration_new(self, event, **post):
+        public_user = request.env.user == request.website.user_id
+        if public_user and event.website_require_login:
+            return request.env["ir.ui.view"].render_template(
```
Comment (yajo) [inline website_event_require_login/controllers/main.py:14]: I think the UI and code could be simplified.

1. Replace the event registration button by a "Login to register" button, which leads directly to `/web/login?whatever`.
1. Once logged in, it returns to the event registration form, but the register button is available now.

**Review by yajo — APPROVED** [review]
This should change, read below.

However it's not bad as it is AFAICS.

**Review by chienandalu — APPROVED** [review]
Functional test ok. What do you think about @Yajo 's comments

**Review by rafaelbn — APPROVED** [review]
:+1:

**Comment by chienandalu** [comment]
The linter is failing on pre-commit when it tries to merge @emagdalenaC2i pass it to your last changes

**Comment by HaraldPanten** [comment]
How's this? Couldn't that be merged? I can see all green @rafaelbn

**Review by HaraldPanten — APPROVED** [review]

**Comment by pedrobaeza** [comment]
A rebase and fix is needed, as this PR is very outdated and have problems with latest pre-commit.

**Comment by HaraldPanten** [comment]
> A rebase and fix is needed, as this PR is very outdated and have problems with latest pre-commit.

Do you recommend us to open a new migration PR? As this one seems to be abandoned...

**Comment by emagdalenaC2i** [comment]
@HaraldPanten Sorry but I can't finish this PR now, so you are welcome to  open a new migration PR

**Comment by HaraldPanten** [comment]
> @HaraldPanten Sorry but I can't finish this PR now, so you are welcome to open a new migration PR

Thx @emagdalenaC2i for your reply. So we'll open a new one 👍

**Comment by rafaelbn** [comment]
Closing as mergin https://github.com/OCA/event/pull/193 with agree of @emagdalenaC2i here: https://github.com/OCA/event/pull/172#issuecomment-705512348

**Pull request closed without merging**